### PR TITLE
Bring back email notifications

### PR DESCRIPTION
**Pull request by asood123**
They were temporarily removed to refactor `transaction.create`. Now that emailLib is cleaner, this PR brings back the notification to all `MEMBER`s of a collective when there is a new transaction.

- Also, adds a `production` check on pushing out to all channels (gitter, slack, email) to avoid spamming

### CODE REVIEW

File: test/notification.model.test.js [hunk 1]
```diff
@@ -30,13 +32,14 @@ var Notification = models.Notification;
 /**
  * Tests.
  */
-describe("notification.model.test.js", () => {
+describe.only("notification.model.test.js", () => {
```
Comment (sedubois) [inline test/notification.model.test.js]: intended?

Comment (asood123) [inline test/notification.model.test.js]: i'll remove it before submitting it... was working with arnaud on the tests

Comment (arnaudbenard) [inline test/notification.model.test.js]: Working on it now

Comment (arnaudbenard) [inline test/notification.model.test.js]: @asood123 I have noticed multiple things in that code:
- The email should be sent with `sendMessageFromActivity` but this function does not exist
- The notifications lib expects `notifConfigs` to be able to send the email. In the tests, this array is empy, I think that's why it doesn't get called

Once those two things are fixed, I think that we will be able to test with a stub.

File: server/lib/utils.js [hunk 2]
```diff
@@ -197,6 +197,22 @@ const defaultHostId = () => {
...
+ * Check if this is an internal email address.
+ * Useful for testing emails in localhost or staging
+ */
+const isEmailInternal = (email) => {
+  if (email.toLowerCase().indexOf('@opencollective.com') != -1 ||
+    email.toLowerCase().indexOf('@opencollective.org') != -1 ||
+    email.toLowerCase() === 'asood123@gmail.com' ||
+    email.toLowerCase() === 'piamancini@gmail.com' ||
+    email.toLowerCase() === 'xdamman@gmail.com' ||
+    email.toLowerCase() === 'sedubois@gmail.com') {
```
Comment (sedubois) [inline server/lib/utils.js]: Could we please avoid committing private emails in source code? If really needed, instead use opencollective.com emails?

Comment (asood123) [inline server/lib/utils.js]: was a bit torn about it. 

Actually, have an idea. I'll wipe the recipient if it's not internal and send the email to the bcc address. That way, we can remove our personal addresses.

File: server/lib/notifications.js [hunk 3]
```diff
@@ -51,7 +58,7 @@ module.exports = (Sequelize, activity) => {
 
 function publishToGitter(activity, notifConfig) {
   const message = activitiesLib.formatMessageForPublicChannel(activity, 'markdown');
-  if (message) {
+  if (message && process.env.NODE_ENV === 'production') {
```
Comment (sedubois) [inline server/lib/notifications.js]: Is this needed? Notify is normally already only called in prod:
https://github.com/OpenCollective/opencollective-api/blob/190521380f85ed80019fef4b2b78d3d84b223706/server/models/Activity.js#L19

File: server/lib/slack.js [hunk 4]
```diff
@@ -29,8 +29,8 @@ module.exports = {
     if (!options) {
       options = {};
     }
-    const message = activitiesLib.formatMessageForPublicChannel(activity, 'slack');
-    return this.postMessage(message, webhookUrl, options);
+      const message = activitiesLib.formatMessageForPublicChannel(activity, 'slack');
+      return this.postMessage(message, webhookUrl, options);
```
Comment (sedubois) [inline server/lib/slack.js]: spurious whitespace?

File: server/lib/slack.js [hunk 5]
```diff
@@ -53,6 +53,11 @@ module.exports = {
     }
 
     return new Promise((resolve, reject) => {
+      // production check
+      if (process.env.NODE_ENV !== 'production') {
```
Comment (sedubois) [inline server/lib/slack.js]: same remark as above

File: test/lib.notifications.test.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+  // stub the transport
+  beforeEach(done => {
+    sinon.stub(nm, 'sendMail', (object, cb) => {
+      cb(null, object);
+    });
+    done();
+  });
+
+  afterEach(done => {
```
Comment (sedubois) [inline test/lib.notifications.test.js]: I guess `afterEach(() => nm.sendMail.restore());` would be sufficient

File: test/lib.notifications.test.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+      user: user,
+      group: group,
+      config: config
+    };
+
+    var subject, body;
+
+    templateData.transaction.id = 1;
+
+    if (templateData.transaction.link.match(/\.pdf$/))
```
Comment (sedubois) [inline test/lib.notifications.test.js]: Missing braces

File: test/lib.notifications.test.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        .send({
+          transaction: transactionsData[0]
+        })
+        .expect(200))
+      .then(res => {
+        expect(res.body).to.have.property('GroupId', group.id);
+        expect(res.body).to.have.property('UserId', user.id); // ...
+      })
+      .then(() => models.Transaction.findAll())
+      .then(transactions => {
```
Comment (sedubois) [inline test/lib.notifications.test.js]: This should be a `tap`

File: test/lib.notifications.test.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+        body = emailLib.getBody(template);
+      })
+      .then(() => request(app)
+        .post('/groups/' + group.id + '/transactions')
+        .set('Authorization', 'Bearer ' + user.jwt(application))
+        .send({
+          transaction: transactionsData[0]
+        })
+        .expect(200))
+      .then(res => {
```
Comment (sedubois) [inline test/lib.notifications.test.js]: This should be a `tap`

File: test/lib.notifications.test.js [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+      })
+      .then(() => models.Transaction.findAll())
+      .then(transactions => {
+        expect(transactions.length).to.equal(1);
+      })
+      .then(() => models.Activity.findAll())
+      .then(activities => {
+        expect(activities.length).to.equal(1);
+      })
+      .then(() => {
```
Comment (sedubois) [inline test/lib.notifications.test.js]: This should be a `tap`

**Pull request merged**